docs/agents/patterns/twitter/twitter_agent.md

```diff
@@ -0,0 +1,341 @@
+    
+    Events are automatically handled by the NEAR AI platform, so you don't need to worry about them.
+
+<hr class="subsection" />
+
+### Processing the Tweet
+
+The agent receives the `tweet` that mentions it as an input object, which contains the following data:
+
+- author_id
+- tweet_id
+- text
+
+Since the `tweet` object contains all the information needed, we can process it through a model in order to generate a response.
+
+``` python title="agent.py"
+# Get the last message, which is the tweet
+last_message = env.list_messages()[-1]
+
+# Check if the last message is None
+if last_message is None:
+    env.add_reply("No message found")
+    print("No message found")
+    return
+
+# Check if the last message is empty
+if not last_message["content"]:
+    env.add_reply("Message content was empty")
+    print("Message content was empty")
+    return
+
+# Get the content of the message
+event = json.loads(last_message["content"])
+
+# Get the text of the tweet
+input_user_text = event["text"]
+# Get the id author of the tweet
+input_tweet_id = event["tweet_id"]
+
+# Generate a prompt
+prompt = [
+    {"role": "system", "content": PROMPT},
+    {"role": "user", "content": input_user_text},
+]
+
+# Generate a response
+joke = env.completion(messages=prompt, model=MODEL)
+
+# Limit the length of the joke to 280 characters
+if len(joke) > 280:
+    joke = joke[:277] + "..."
+```
+
+<hr class="subsection" />
+
+### Answering
+
+The agent uses the `tweepy` library to send a reply to the tweet.
+
+First it creates a `tweepy` client, which is done in the `__init__` method of the agent. The client is created using the API keys stored in the environment variables.
+
+``` python title="agent.py"
+self.x_client = tweepy.Client(
+          consumer_key=self.x_consumer_key,
+          consumer_secret=self.x_consumer_secret,
+          access_token=self.x_access_token,
+          access_token_secret=self.x_access_token_secret
+      )
+```
+
+Then, it uses the `send_tweet` method - which takes the `env`, `tweet`, and `tweet_id` - to send a reply to the tweet.
+
+``` python title="agent.py"
+
+def send_tweet(self, env, tweet, tweet_id):
+    env.add_reply(f"Sending tweet: {tweet}")
+    response = self.x_client.create_tweet(text=tweet, in_reply_to_tweet_id=tweet_id)
+    print(f"Tweet published! ID: {response.data['id']}")
+
+    # Send the tweet
+    self.send_tweet(env, joke, input_tweet_id)
+```
+
+----
+
+## Modifying the Agent
+
+If you want to create your own agent, you will need to start by forking [maguila](https://app.near.ai/agents/maguila.near/jokester/latest/source), and setting up the right Twitter API keys, so the agent can control the account you want.
+
+<hr class="subsection" />
+
+### Forking an Agent
+
+To fork the Twitter agent, enter [click here](https://app.near.ai/agents/maguila.near/jokester/latest/source) in your browser and click on the "Fork" button. This will create a copy of the agent in your NEAR AI registry.
```

Comment:
```suggestion
To fork the Twitter agent, [open this link](https://app.near.ai/agents/maguila.near/jokester/latest/source) in your browser and click the **`Fork`** button. This will create a copy of the agent in your NEAR AI registry.
```